### PR TITLE
2376 torch 1.9 compatibility

### LINKED ISSUE #2376: compatibility with torch 1.9
**Is your feature request related to a problem? Please describe.**
pytorch 1.9 is available, there're some compatibility issues such as https://github.com/Project-MONAI/MONAI/runs/2831507688?check_suite_focus=true#step:8:4236

### PR DESCRIPTION
**Pull request by wyli**
Fixes #2376

### Description
- fixes unit/integration tests
- adds torch 1.9 tests

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [x] In-line docstrings updated.

### CODE REVIEW

File: .github/workflows/pythonapp.yml [hunk 1]
```diff
@@ -205,16 +205,13 @@ jobs:
     strategy:
       matrix:
         environment:
-          - "PT16+CUDA102"
```
Comment (Nic-Ma) [inline .github/workflows/pythonapp.yml:208]: Hi @wyli , thanks for the quick fix, may I know the strategy of PT tests? Only maintain the latest 6 environments?

Thanks.

**Comment by wyli** [comment]
tested 
- https://github.com/Project-MONAI/MONAI/actions/runs/941035540
- https://github.com/Project-MONAI/MONAI/actions/runs/940931971

File: .github/workflows/pythonapp.yml (hunk 1 above)
Comment (wyli) [reply .github/workflows/pythonapp.yml:208]: the premerge ones test both docker + pip version, currently it's set to run 6 different versions

Comment (Nic-Ma) [reply .github/workflows/pythonapp.yml:208]: I see.

Thanks.

**Review by Nic-Ma — APPROVED** [review]

**Pull request merged**